src/math.rs

```diff
@@ -16,11 +16,59 @@ use cgmath::{
     BaseFloat, Decomposed, EuclideanSpace, InnerSpace, Point3, Quaternion, Rotation, Vector2,
     Vector3,
 };
-use collision::{Aabb, Aabb3};
+use collision::{Aabb, Aabb3, Contains};
 use num_traits::identities::One;
 use num_traits::Float;
+use std::convert::From;
+use std::fmt::Debug;
 use std::ops::Mul;
 
+pub trait PointCulling<S>: objekt::Clone + Debug + Sync + Send
+where
+    S: BaseFloat + Sync + Send,
+{
+    fn contains(&self, point: &Point3<S>) -> bool;
+    fn intersects(&self, aabb: &Aabb3<S>) -> bool;
+    fn transform(&self, isometry: &Isometry3<S>) -> Box<PointCulling<S>>;
```

Comment:
This can probably be impl PointCulling, I'd think.

Reply:
Unfortunately this does not work inside a trait, from [rfcs](https://github.com/rust-lang/rfcs/blob/master/text/1522-conservative-impl-trait.md)

_"impl Trait may only be written within the return type of a freestanding or inherent-impl function, not in trait definitions or any non-return type position. "_